### ISSUE TITLE
sdl window.render is consuming tons of memory in a typical render loop

### ISSUE THREAD

**Issue by mreinstein** [kmamal/node-sdl#47]
These particular lines:

```javascript
const buffer = canvas.toBuffer('raw')
window.render(width, height, width * 4, 'bgra32', buffer)
```
(from https://github.com/kmamal/node-sdl/blob/master/examples/03-canvas-drawing/index.js#L17-L18 but present in most raw drawing examples)

When this is executed in a draw loop, within 20 minutes it exhausts all the memory in my system.  when I comment out _only_ the `window.render` line the memory leak is gone

@kmamal  my uneducated theory is this call is holding a reference to the `buffer` object and not releasing it, preventing normal garbage collection

**Comment by kmamal** [comment]
Very likely. I might have missed something during the big c++ rewrite I did. I'll have a look.

**Comment by kmamal** [comment]
Should be fixed in `v0.10.2`. As you said, I forgot to free a texture.

**Comment by mreinstein** [comment]
confirmed fixed, thanks so much!!!
